src/hamster/lib/datetime.py

```diff
@@ -590,25 +590,33 @@ def from_start_end(cls, start, end=None):
         Convenient to ease backward compatibility,
         and to handle either hdays or datetimes.
         """
+
         if isinstance(start, Range):
             assert end is None, "range and end are mutually exclusive"
             range = start
-        else:
```

Comment:
You moved a lot of code out of this `else`, but is that really needed? Even more, doesn't this break passing a `Range` object? AFAICS, the line

    range = Range(start, end)

now runs even when a `Range` is passed, overwriting the `range = start` above?

Reply:
Sure, I'm downright ashamed about this commit.
The first if should return.

It was a sunday evening quick fix, and it looks like I'm no longer into it.
I'll try to fix it this evening.